sources/core/objects/turfs/MetalWall.cpp

```diff
@@ -37,6 +41,68 @@ void MetalWall::AttackBy(IdPtr<Item> item)
     }
 }
 
+void MetalWall::AfterWorldCreation()
+{
+    UpdateState(Dir::ALL);
+}
+
+void MetalWall::Delete()
+{
+    UpdateNeighborhoodState();
```

Comment:
I think that won't work. When `Delete` is called the wall still exists on tile, so here some workaround is needed.
Some example can be found in the table implementation - there is a similar functionality.
Basic approach is to add `ignored_wall` id parameter in the  `UpdateNeighborhoodState();` and pass deleted wall into it - then it will be possible to ignore walls if they have `ignored_wall` id.

Reply:
Well, it is pretty strange that it works. I will check soon why the fuck it works - seems like I'm missing something really important here.